### DISCUSSION TITLE
Debian runtime via WebVM

### DISCUSSION THREAD

**Discussion by aehlke** [live-codes/livecodes#435] [Ideas]
**[WebVM](https://leaningtech.com/webvm-server-less-x86-virtual-machines-in-the-browser/)**

Looks promising for running any compiler that can be hosted inside this debian-in-wasm runtime. Maybe a way to run swift compiler, for example, or to get more reuse out of build toolchains for linux env

Demo: https://webvm.io/
Article: https://leaningtech.com/webvm-server-less-x86-virtual-machines-in-the-browser/

**Comment by hatemhosny** [comment]
this approach is probably promising. It can open a wide variety of additional language support.

An open source alternative: https://github.com/copy/v86

**Reply by aehlke** [reply]
I believe WebVM is also open source, with Apache2: https://github.com/leaningtech/webvm by being Debian it might already have strong build capabilities

**Reply by hatemhosny** [reply]
> WebVM depends on the CheerpX x86-to-WebAssembly virtualization technology.
>
> The public CheerpX deployment is provided as-is and is free to use for technological exploration, testing and non-commercial uses. Downloading a CheerpX build for the purpose of hosting it elsewhere is not permitted.
>
> If you want to build a product on top of CheerpX/WebVM, please get in touch

https://github.com/leaningtech/webvm#license
